Cleanup tempfiles

zhijianli88 opened the pull request:
After the PR, we can cleanup all remaining tempdir

lizj@FNSTPC:~/workspace/k8s/helm$ export TMPDIR=/tmp/xxxxxx ; make clean; make test >/dev/null && echo PASS; ls -l $TMPDIR
PASS
total 0

zhijianli88 committed:
testing: Use T.cleanup() to cleanup tempdir

Previously, after 'make test', some many temp dir have not been removed.
Since this commit, temp dir created by ensure.HelmHome()
ensure.TempDir() and ensure.TempFile() will be removed automatically in
the end of testing

- Use T.cleanup to remove the tempdir
- Remove unnecessary defer

Signed-off-by: Li Zhijian <lizhijian@cn.fujitsu.com>

zhijianli88 committed:
testing: setup a temp HelmHome for a few tests

these tests will dirty the HelmHome

Signed-off-by: Li Zhijian <lizhijian@cn.fujitsu.com>

provPaulBrousseau commented on internal/test/ensure/ensure.go:
```diff
@@ -36,9 +36,6 @@ func HelmHome(t *testing.T) func() {
 	os.Setenv(helmpath.ConfigHomeEnvVar, "")
 	os.Setenv(helmpath.DataHomeEnvVar, "")
-	return func() {
-		os.RemoveAll(base)
```
It might be useful to leave the directory in place if the test fails.

zhijianli88 replied:
indeed, but currently, the HelmHome() caller always call it as 'defer HelmHome(t)() ' which has the same behaviors with T.cleanup() i guess.

provPaulBrousseau replied:
Apologies; I wasn't clear with my comment.  I don't think that `helm` should keep the returned func (I like how it's cleaned up), but rather, in the func called by `t.Cleanup`, it might be good to check `t.Failed` before calling `os.RemoveAll`, because if a test fails, it could be useful to look in those directories for the state of config, cache, etc.  Maybe that's a different PR & discussion.

zhijianli88 replied:
thanks for your explanation, i just append a commit to do this

zhijianli88 committed:
testing: Don't cleanup tempfiles if the test failed

As suggestion by Paul Brousseau, It might be useful to leave the directory in
place if the test fails.

Signed-off-by: Li Zhijian <lizhijian@cn.fujitsu.com>

zhijianli88 commented:
Kindly ping

zhijianli88 commented:
i just rebased the PR, anyone could help to have a review ?

zhijianli88 commented:
Ping
